cishell/src/main/java/org/cishell/cibridge/cishell/impl/CIBridgeGUIBuilderService.java

```diff
@@ -1,64 +1,288 @@
+                        null, false, false);
+            } else {
+                notification = new Notification(id, NotificationType.FORM, null, null,
+                        null, null, null, false,
+                        null, false, false);
+            }
+
+            gui = createAndGetGui(id, map, notification);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return gui;
     }
 
     @Override
     public Dictionary createGUIandWait(String id, MetaTypeProvider params) {
-        return null;
+
+        HashMap<String, Notification> map = notificationFacade.getNotificationMap();
+
+        ObjectClassDefinition objectClassDefinition = params.getObjectClassDefinition(id, null);
+        List<ParameterDefinition> notificationParams = new ArrayList<>();
+
+        for (AttributeDefinition attributes : objectClassDefinition.getAttributeDefinitions(ObjectClassDefinition.ALL)) {
+            List<Property> properties = new ArrayList<>();
+            for (int i = 0; i < attributes.getOptionLabels().length; i++) {
+                Property property = new Property(attributes.getOptionLabels()[i], attributes.getOptionValues()[i]);
+                properties.add(property);
+            }
+            //TODO fill parameter definition fields using setters
+            ParameterDefinition parameterDefinition = new ParameterDefinition(attributes.getID());
+            notificationParams.add(parameterDefinition);
+        }
+
+        Notification notification = new Notification(id, NotificationType.FORM, objectClassDefinition.getID(), objectClassDefinition.getName(),
+                objectClassDefinition.getDescription(), null, notificationParams, false,
+                null, false, false);
+
+        GUI gui = createAndGetGui(id, map, notification);
+
+        return gui.openAndWait();
     }
 
     @Override
     public boolean showConfirm(String title, String message, String detail) {
-        return false;
+
+        HashMap<String, Notification> map = notificationFacade.getNotificationMap();
+
+        String UUID = counter.randomUUID().toString();
+
+        Notification notification = new Notification(UUID, NotificationType.CONFIRM, title, message,
+                detail, null, null, false,
+                null, false, false);
+
+        GUI gui = createAndGetGui(UUID, map, notification);
+        gui.openAndWait();
+
+        return notification.getConfirmationResponse();
     }
 
     @Override
     public void showError(String title, String message, String detail) {
 
+        HashMap<String, Notification> map = notificationFacade.getNotificationMap();
+
+        String UUID = counter.randomUUID().toString();
+
+        Notification notification = new Notification(UUID, NotificationType.ERROR, title, message,
+                detail, null, null, false,
+                null, false, false);
+
+        map.put(UUID, notification);
```

Comment:
This and the next line should really be encapsulated in the facade. You shouldn't be using the 'notification map' as this could cause unintended consequences down the line. Instead add an `addNotification(notification)` (notification.getID() can be used to get the id in the function) method to the facade (not to graphql) which does exactly what these lines do and block off access to the notification map. Notification map is an implementation detail.

Replace all instances of this pattern with `addNotification`